src/Evolution/DiscontinuousGalerkin/MortarData.hpp

```diff
@@ -0,0 +1,117 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <optional>
+#include <ostream>
+#include <pup.h>
+#include <utility>
+#include <vector>
+
+#include "ErrorHandling/Assert.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Parallel/PupStlCpp17.hpp"
+#include "Time/TimeStepId.hpp"
+
+namespace evolution::dg {
+/*!
+ * \brief Data on the mortar used to compute the boundary correction for the
+ * DG scheme.
+ *
+ * The class holds the local data that has been projected to the mortar as well
+ * as the neighbor data that has been projected to the mortar. The local and
+ * neighbor data is later used to compute the same unique boundary correction on
+ * the mortar for both elements. That is, the final boundary correction
+ * computation is done twice: one on each element touching the mortar. However,
+ * the computation is done in such a way that the results agree.
+ *
+ * In addition to the (type-erased) fields on both sides of the mortar, the face
+ * (not mortar!) mesh of the neighbor is stored. The mesh will be necessary
+ * when hybridizing DG with finite difference or finite volume schemes
+ * (DG-subcell).
+ *
+ * If the element and its neighbor have unaligned logical coordinate systems
+ * then the data is stored in the local logical coordinate's orientation
+ * (\f$\xi\f$ varies fastest). This means the action sending the data is
+ * responsible for reorienting the data on the mortar so it matches the
+ * neighbor's orientation.
+ *
+ * \tparam Dim the volume dimension of the mesh
+ */
+template <size_t Dim>
+class MortarData {
+ public:
+  /*!
+   * \brief Insert data onto the mortar. This function must be called once
+   * between calls to extract.
+   *
+   * The insert functions require that:
+   * - the data is inserted only once
```

Comment:
Since with the Doxygen comment is applied once to each function I'm not clear it's confusing. I did add a sentence before the bullet list `Exactly one local and neighbor insert call must be made between calls to extract`. What are your thoughts?

Reply:
To me this sentence clarified things nicely, thank you 🙂